src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -210,7 +210,11 @@ internal static byte[] GetEncodedTruncatedBytesFromString(string? text, Encoding
                 return encoding.GetBytes(text);
             }
 
```

Comment:
Is it common for long text to be used?  maxBytes is either going to be 0 or 65K, and 0 is handled by the above path.  Rather than bothering with the ArrayPool, would it make sense instead to just add an additional special-case to the above:
```C#
if (maxBytes == 0 || encoding.GetMaxByteCount(text.Length) <= maxBytes)
{
    return encoding.GetBytes(text);
}
```
For UTF8, then, we'd be even faster for text up to 21K chars, which naively seems like it'd be the vast majority case, and then only have to fall back to the slower path (which is already pretty slow for > 21K chars as it's enumerating each rune individually).

Is there a scenario that drove you to want to use ArrayPool here?